src/main/proto/ga4gh/read_service.proto

```diff
@@ -91,6 +99,49 @@ message GetReadGroupSetRequest {
   string read_group_set_id = 1;
 }
 
+// ******************  /readgroups  *********************/
+
+// This request maps to the body of `POST /readgroups/search` as JSON.
+
+
+message SearchReadGroupsRequest {
+  // The dataset to search (required).
+  string dataset_id = 1;
```

Comment:
We need to hold off on this change until we work through (and
document) the use cases.

Obtaining all readgroups in a dataset (project) is something
that one would almost never want to do.

In my experience, the command analysis work flow is:
   1) Find BAMs (readgroupset) meeting meeting some criteria
      defined by the metadata.  It many contexts, this is more
      refined than sample.  For instance, a sample can have both
      RNA-Seq and genome sequencing.
   2) retrieve reads from the readgroup, regardless of readgroup set.
      

It would simplified the API if we didn't push dataset down to
the lower levels unless there is a compelling use case and stick
to the container hierarchy.  The container hierarchy will serve
the `find all' type of queries (walking the hierarchy) and
criteria queries can build in this plus the metadata.

Again, we are up against dataset being an unclear concept.




Jerome Kelleher <notifications@github.com> writes:
> In src/main/proto/ga4gh/read_service.proto:
> 
> > @@ -91,6 +99,49 @@ message GetReadGroupSetRequest {
> >    string read_group_set_id = 1;
> >  }
> >
> > +// ******************  /readgroups  *********************/
> > +
> > +// This request maps to the body of `POST /readgroups/search` as JSON.
> > +
> > +
> > +message SearchReadGroupsRequest {
> > +  // The dataset to search (required).
> > +  string dataset_id = 1;
> 
> I think we should add back in read_group_set_id (optional). I'm thinking of the
> case here where we need to find the ReadGroups that correspond to a particular
> ReadGroupSet so that the client can run a SearchReadsRequest. With this
> inferface, the client will have to get back all of the ReadGroups in a dataset
> (potentially hundreds of thousands), and then filter for the read group set
> they want.
> 
> I guess the alternative to this is to add a read_group_set option to
> SearchReadsRequest to cover this use-case. Since Google already does this
> (AFAIK), this might be the simpler option.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.*
>